src/main/java/net/javadiscord/javabot/systems/moderation/BanCommand.java

```diff
@@ -6,27 +6,68 @@
 import net.dv8tion.jda.api.requests.restaction.interactions.ReplyCallbackAction;
 import net.javadiscord.javabot.command.ResponseException;
 import net.javadiscord.javabot.command.Responses;
-import net.javadiscord.javabot.command.moderation.ModerateUserCommand;
+import net.javadiscord.javabot.command.moderation.ModerateCommand;
+import net.javadiscord.javabot.command.moderation.UserPermissionException;
 
 /**
  * Command that allows staff-members to ban guild members.
  */
-public class BanCommand extends ModerateUserCommand {
+public class BanCommand extends ModerateCommand {
 
 	@Override
-	protected ReplyCallbackAction handleModerationActionCommand(SlashCommandInteractionEvent event, Member commandUser, Member target) throws ResponseException {
+	protected ReplyCallbackAction handleModerationCommand(SlashCommandInteractionEvent event, Member commandUser) throws ResponseException {
 		var reasonOption = event.getOption("reason");
-		if (reasonOption == null) {
+		var targetOption = event.getOption("user");
+
+		if (reasonOption == null || targetOption == null) {
 			return Responses.error(event, "Missing required arguments.");
 		}
-		var reason = reasonOption.getAsString();
+
+		String reason = reasonOption.getAsString();
+		Member target = targetOption.getAsMember();
 		boolean quiet = event.getOption("quiet", false, OptionMapping::getAsBoolean);
 
-		var moderationService = new ModerationService(event.getInteraction());
-		if (moderationService.ban(target, reason, commandUser, event.getTextChannel(), quiet)) {
-			return Responses.success(event, "User Banned", String.format("%s has been banned.", target.getAsMention()));
-		} else {
-			return Responses.warning(event, "You're not permitted to ban this user.");
+		if (target == null) {
+			String snowflakeID = targetOption.getAsString();
+			return handleAsID(event, commandUser, snowflakeID, reason, quiet);
 		}
+
+		if (target.getId().equals(commandUser.getId())) {
+			Responses.error(event, "You cannot ban yourself.");
+		}
+
+		if (event.getJDA().getSelfUser().getId().equals(target.getId())) {
+			Responses.error(event, "You cannot ban the bot.");
+		}
+
+		return handleAsMember(event, commandUser, target, reason, quiet);
+	}
+
+	private ReplyCallbackAction handleAsID(SlashCommandInteractionEvent event, Member commandUser, String snowflakeID, String reason, boolean quiet) {
+		ModerationService moderationService = new ModerationService(event.getInteraction());
+
+		try {
+			moderationService.ban(snowflakeID, reason, commandUser, event.getTextChannel(), quiet);
+		} catch (IllegalArgumentException e) {
+			return Responses.error(event, "Incorrect Argument");
+		} catch (UserPermissionException e) {
+			return Responses.error(event, e.toString());
+		}
+
+		return Responses.success(event, "User Banned", String.format("%s has been banned.", snowflakeID));
+	}
```

Comment:
Just keep this method (or simply merge it with the handleModerationCommand method above) and remove the other one. There's no need to have an extra method for a member object if we can just use `Member#getIdLong`
However, rename the `snowflakeID` to `targetId` and make it a long instead of a String.

Reply:
This method does not check for much since it's using the ID and has no garantee that the user exists within the server whereas, member does. the user option makes sure that the user is in the server (if it returns a member)

Reply:
The entire point of this PR is to get rid of that check, no?